### ISSUE TITLE
Error: invalid_grant after successful authorization

### ISSUE THREAD

**Issue by natalyjazzviolin** [msigwart/gatsby-source-google-calendar#7]
The plugin does not work after it's successfully authorized. I'm placing all tokens (ID, SECRET, ACCESS, REFRESH) in both the .end.development and .env.production files.

This is the message I get:
```
File: node_modules/gatsby-source-google-calendar/gatsby-node.js:208:11

Error: invalid_grant
```
Is there a specific way I should be configuring the OAuth Consent Screen? I'd appreciate any advice, thank you in advance!

**Comment by msigwart** [comment]
Thanks for the issue! I've googled a bit, unfortunately it's a pretty generic error message... I guess you added `http://localhost:8000/oAuthCallback` as redirect URI when setting up the OAuth Client? 

Also, some of the following threads may be able to help you out:
* https://github.com/googleapis/google-api-nodejs-client/issues/1951
* https://stackoverflow.com/questions/10576386/invalid-grant-trying-to-get-oauth-token-from-google

Maybe you could also try it with another project and see if the error happens again.

**Comment by clariceabreu** [comment]
I solved this but opening the authentication URL again. I think perhaps you need to refresh the token.
You can access:
```https://accounts.google.com/o/oauth2/v2/auth?access_type=offline&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcalendar.readonly&response_type=code&client_id=[YOUR-CLIENT-ID]&redirect_uri=http%3A%2F%2Flocalhost%3A8000%2FoAuthCallback```
And update the env vars with the new result

**Comment by msigwart** [comment]
Glad you were able to sort out the issue, I'm gonna close this issue for now.

**Comment by Carlo-Piantini** [comment]
Can we re-open this issue? I'm still getting the same error, regardless of how many times I'm refreshing the authorization token via the authorization URL. I followed the [documentation](https://www.gatsbyjs.com/plugins/gatsby-source-google-calendar/) to setup the plugin. Here's the full error message:
```
 ERROR #11321  PLUGIN

"gatsby-source-google-calendar" threw an error while running the sourceNodes lifecycle:

invalid_grant

  202 |   const calendar = google.calendar({version: 'v3', auth});
  203 |   if (calendarIds.length) {
> 204 |     return await Promise.all(calendarIds.map(async calendarId => (await calendar.calendarList.get({
      |                                                                   ^
  205 |       calendarId
  206 |     })).data));
  207 |   }

File: node_modules/gatsby-source-google-calendar/gatsby-node.js:204:67



  Error: invalid_grant
  
  - gaxios.ts:158 Gaxios._request
    [capitol-city-church]/[gaxios]/src/gaxios.ts:158:15
  
  - task_queues.js:95 processTicksAndRejections
    internal/process/task_queues.js:95:5
  
  - oauth2client.js:172 OAuth2Client.refreshTokenNoCache
    [capitol-city-church]/[google-auth-library]/build/src/auth/oauth2client.js:172:21
  
  - oauth2client.js:196 OAuth2Client.refreshAccessTokenAsync
    [capitol-city-church]/[google-auth-library]/build/src/auth/oauth2client.js:196:19
  
  - oauth2client.js:367 OAuth2Client.requestAsync
    [capitol-city-church]/[google-auth-library]/build/src/auth/oauth2client.js:367:21
  
  - gatsby-node.js:204 
    [capitol-city-church]/[gatsby-source-google-calendar]/gatsby-node.js:204:67
  
  - async Promise.all
  
  - gatsby-node.js:204 getCalendars
    [capitol-city-church]/[gatsby-source-google-calendar]/gatsby-node.js:204:12
  
  - gatsby-node.js:76 Object.exports.sourceNodes
    [capitol-city-church]/[gatsby-source-google-calendar]/gatsby-node.js:76:21
  
  - api-runner-node.js:429 runAPI
    [capitol-city-church]/[gatsby]/src/utils/api-runner-node.js:429:16
```

**Comment by msigwart** [comment]
Hm, okay, thanks for letting me know, I'll reopen the issue. When I get a chance, I'll look into it again.

**Comment by ronalde** [comment]
FYI: in my case I got this error due to copy-pasting the newlines of the values of the  `GOOGLE_ACCESS_TOKEN` and `GOOGLE_REFRESH_TOKEN` fields (I use `urxvt`). Field and value (`GOOGLE_ACCESS_TOKEN=....`) shoud be one a single line. 

So the solution would be to stipulate that in the documentation:
```diff
--- node_modules/gatsby-source-google-calendar/README.md	2021-10-15 18:59:09.982245182 +0200
+++ /tmp/README.md	2021-10-16 13:24:23.922243936 +0200
@@ -99,8 +99,9 @@
-Store these values in your `.env` files, then restart `gatsby develop`.
-The plugin should now query the events from Google Calendar.
+Store these lines, eg. the complete `FIELD=VALUE` lines without
+newlines, in your `.env` files, then restart `gatsby develop`.  The
+plugin should now query the events from Google Calendar.
```

**Comment by msigwart** [comment]
> FYI: in my case I got this error due to copy-pasting the newlines of the values of the GOOGLE_ACCESS_TOKEN and GOOGLE_REFRESH_TOKEN fields (I use urxvt). Field and value (GOOGLE_ACCESS_TOKEN=....) shoud be one a single line.

Funny enough, I just made the same mistake trying to debug this issue 😅

I've updated the documentation accordingly and also added a [troubleshooting section](https://github.com/msigwart/gatsby-source-google-calendar#troubleshooting) that describes a few things that can be done to resolve this issue.

In the future, I might add the option to query calendar events using an API key to skip the whole authentication process. This would work only with public calendars though.
